### ISSUE TITLE
Previews failing with 500 error

### ISSUE THREAD

**Issue by lusa** [birkir/gatsby-source-prismic-graphql#39]
Hello, Previews have been working great with this plugin, but now they are no longer working for our team. I get a 500 error on the fetch to get the preview, a notice that it has been blocked by CORS policy. We haven't updated anything recently, and it has been working fine with Gatsby and Prismic. Any ideas?

```
GET ... 500 (Internal Server Error)

Access to fetch at 'https://xxxx.prismic.io/graphql?query=...from origin 'http://localhost:8000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Error: Network error: Failed to fetch
    at new ApolloError
```

**Comment by ancashoria** [comment]
I was trying to setup previews and I get the same error.

**Comment by kkor** [comment]
Same here!

**Comment by xvladghidiu** [comment]
+1

**Comment by srenault** [comment]
Hello,

I'm working at Prismic. I wanted to let you know that this issue is not related to the gatsby-source-prismic-graphql plugin but an issue on our side. It doesn't affect all draft previews. We're currently in the process of deploying the fix for everyone. It should be good for everybody at the end of the week.

Sorry for the inconvenience.

**Comment by trickydisco78** [comment]
Yes i'm getting this issue as well. I need to make an assessment on whether we use this new prismic graphql plugin or revert back to the old one and with a lot more code.

**Comment by srenault** [comment]
Hello again, everything should be back to normal. We deployed the fix on our side and @birkir published a new version of `gatsby-source-prismic-graphql`. Please let me know if you still have trouble with preview.

**Comment by trickydisco78** [comment]
Excellent. @birkir  Is it out of alpha now then?

**Comment by trickydisco78** [comment]
I'm still getting a server 500 error
![Screenshot 2019-06-05 at 09 41 06](https://user-images.githubusercontent.com/3627429/58942654-15043a00-8776-11e9-823b-7e7a10a65882.png)

```https://greatstate.prismic.io/graphql?query=query%20PrismicBlogArticleByDocumentID(%24uid%3A%20String!)%20%7B%20blog_article(uid%3A%20%24uid%2C%20lang%3A%20%22en-gb%22)%20%7B%20...prismicBlogArticleFields%20__typename%20%7D%20%7D%20&operationName=PrismicBlogArticleByDocumentID&variables=%7B%22uid%22%3A%22sitecore-introduces-ai-and-machine-learning%22%2C%22id%22%3A%22XJJSahAAAGiVWAwt%22%2C%22lang%22%3A%22en-gb%22%7D```

**Comment by trickydisco78** [comment]
Looks like i'm getting an unknown fragment. Looks like it can't resolve fragments? How does prismic resolve fragments from the website?

**Comment by trickydisco78** [comment]
To clarify it looks like the cors issue has been fixed. Looks like i'm having a separate issue about using fragments that the api can consume.
Ah ha.. got it working. I was missing ```BlogArticle.fragments = [prismicBlogArticleFields]```

**Comment by birkir** [comment]
You have to include the fragment source in the query

const query = graphql`
  query {
    ...MyFragment
  }
  ${MyFragment}
`;

**Comment by birkir** [comment]
Great! I think we can close this issue now.

**Comment by codebravotech** [comment]
@trickydisco78 Hod did you get this working? It fixes the 500 when i declare the fragments array on my component, but I have to put them in quotes or its an unknown variable (I think that's a typo in your response above) and if I do that, it combines all of the fragment names into one, so I get an "Unexpected name" graphql query error.

Additionally, if I just drop it down to one fragment name (the one that was causing the 500), it still succeeds but tells me "Unexpected Name" graphql query error with just that name. I'm tearing my hair out here, please help!

**Comment by codebravotech** [comment]
@birkir Please reopen? since this is cropping up for other users also and there's no complete explanation of a solution/workaround
